### PR TITLE
Issue/refactor developer docs

### PR DESCRIPTION
**Pull request by wouterdb**
# Description

Refactored docs as a preparation for modules V2

```
Environment Variabes -> Move to Administrator Documentation


Model developer documentation -> add short intro pragraph 
- Developer Getting Started  -> Needs to be reviewed
- Project Creation Guide
- Module Creation Guide  -> split from `Module Developers Guide` (+V2)
- Understanding Modules -> split from `Module Developers Guide` (+V2)
- Developing Plugins -> split from `Module Developers Guide` 
- Developing South Bound Integrations  -> split from `Module Developers Guide` 
- Releaseing Modules -> split from `Module Developers Guide`  (+V2)
- Testing Modules -> rename of `Test plugins`	
- Debugging Modules -> rename of `Model debugging`	
- Model Design Guideline
```
# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [ ] ~~Attached issue to pull request~~
- [x] Changelog entry
- [ ] ~~Type annotations are present~~
- [ ] ~~Code is clear and sufficiently documented~~
- [ ] ~~No (preventable) type errors (check using make mypy or make mypy-diff)~~
- [ ] ~~Sufficient test cases (reproduces the bug/tests the requested feature)~~
- [ ] ~~Correct, in line with design~~
- [x] End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )

### CODE REVIEW

File: docs/model_developers/developer_getting_started.rst [hunk 1]
```diff
@@ -108,7 +108,7 @@ When working on an existing project, you need to ``clone`` them first:
 
 .. code-block:: bash
 
-    git clone project_name
+    git clone project_url
```
Comment (arnaudsjs) [inline docs/model_developers/developer_getting_started.rst:111]: This makes the notation consistent with the `cd` command mentioned above.
```suggestion
    git clone <project_url>
```

File: docs/model_developers/developer_getting_started.rst [hunk 2]
```diff
@@ -76,30 +80,26 @@ To create a new project:
...
-    pip install inmanta-core
-
-    pip install pytest
+    pip install inmanta-core pytest-inmanta
 
-    pip install pytest-inmanta
 
-
-Once you are done with creating a project and installing the required modules, you can ``cd`` into that directory and open vs code by running:
+Once you are done with creating a project, you can ``cd`` into that directory and open vs code by running:
```
Comment (arnaudsjs) [inline docs/model_developers/developer_getting_started.rst:92]:
```suggestion
Once you are done with creating a project, you can ``cd`` into that directory and open VS Code by running:
```

File: docs/model_developers/developer_getting_started.rst [hunk 3]
```diff
@@ -173,40 +174,49 @@ Same as :ref:`Working on a New Project` part, modules can also be created like:
...
+    mkdir -p $INMANTA_TEST_ENV
+
 
-1. Create the required TEST directories:
+2. Install required dependencies
 
 .. code-block:: bash
 
-    mkdir -p /tmp/env
+    pip install -r requirements.txt requirements.dev.txt
```
Comment (arnaudsjs) [inline docs/model_developers/developer_getting_started.rst:216]:
```suggestion
    pip install -r requirements.txt -r requirements.dev.txt
```

**Review by arnaudsjs — CHANGES_REQUESTED** [review]

**Review by Pouriya-Inmanta — APPROVED** [review]

**Review by sanderr — CHANGES_REQUESTED** [review]

File: docs/model_developers/developer_getting_started.rst [hunk 4]
```diff
@@ -173,40 +174,49 @@ Same as :ref:`Working on a New Project` part, modules can also be created like:
...
+    mkdir -p $INMANTA_TEST_ENV
+
 
-1. Create the required TEST directories:
+2. Install required dependencies
 
 .. code-block:: bash
 
-    mkdir -p /tmp/env
+    pip install -r requirements.txt -r requirements.dev.txt
```
[the remark this answers is not in the document]
Comment (wouterdb) [reply docs/model_developers/developer_getting_started.rst:216]: yes, the test_env is used by the compiler, it should manage itself. 

This is for the venv where the tests are run in (and the can be different)

We could indeed also point INMANTA_TEST_ENV to the developer venv

[the remark this answers is not in the document]
Comment (sanderr) [reply docs/model_developers/developer_getting_started.rst:215]: Of course, I see. We could point the test env there but I don't think we should in general. I'm in favor of using a tempdir for the documention.

**Review by sanderr — APPROVED** [review]

**Review by bartv — APPROVED** [review]

**Review by arnaudsjs — APPROVED** [review]

**Pull request closed without merging**